tests/incremental/derive_macro_expansion/proc_macro_unchanged.rs

```diff
@@ -0,0 +1,37 @@
+// This test tests that derive-macro execution is cached.
+// HOWEVER, this test can currently only be checked manually,
+// by running it (through compiletest) with `-- --nocapture --verbose`.
+// The proc-macro (for `Nothing`) prints a message to stderr when invoked,
+// and this message should only be present during the second invocation
+// (which has `cfail2` set via cfg).
+// FIXME(pr-time): Properly have the test check this, but how? UI-test that tests for `.stderr`?
```

Comment:
Yep, that's what the "Properly have the test check this, but how?" was for, I simply didn't see how I can check this. There are directives, but I didn't figure out a way to make them work for this case. Because the directive needs to be attached to the proc macro output I think, but that was weird somehow. I will try again and see what exactly the problem was with that.

Reply:
Tried again, but couldn't figure out something that works, i.e., that succeeds with proc-macro caching enabled, but fails with proc-macro caching disabled.